### ISSUE TITLE
Allow async-await rule to pass when line contains "return fnAsync()"

### ISSUE THREAD

**Issue by bompus** [IPWright83/eslint-plugin-async-protect#3]
Your rules are awesome, but I keep finding myself disabling them more than I should need to. I believe the code sample below will explain the problem:

```
async routeTypeUserListAsync(req) {
   // eslint-disable-next-line async-protect/async-await
   return runner.runTypeListAsync(req, this);
}
```

So obviously, I want to mark both asynchronous functions as such with the Async suffix, makes sense. However, being forced to add an additional "await" to pass isn't good for performance. This is a very common pattern in async-await usage.

I'm sure that you're aware, but for others seeing this, each time you "await" something, it creates a new promise that has to resolve, wasting CPU cycles.

Most people would just give up and use "return await runner.runTypeListAsync(req, this);" , but this adds an extra promise into the mix for not reason, when any async function should be able to simply return another async function (or promise).

Does this make sense to you?

[Further reading: eslint no-return-await rule](https://eslint.org/docs/rules/no-return-await)
[Further reading: why using 'return await' is a bad idea](http://hassansin.github.io/Why-return-await-Is-a-Bad-Idea)

**Comment by IPWright83** [comment]
@bompus for some reason I've only just spotted this, so sorry for not replying sooner. I'm glad you find the plugin helpful. That makes complete sense, and I can see you've put together a PR. I'll take a look at that (hopefully tonight) and provide any feedback/or merge it in and re-publish.

**Comment by IPWright83** [comment]
@bompus this is all up and ready, published as `eslint-plugin-async-protect@1.1.0`, didn't quite warrant a major version bump.

**Comment by bompus** [comment]
@IPWright83 Thanks so much for reviewing it quickly.
